wasm crash on window.SetContent with 2.6.0-beta1

### Checklist

- [x] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [x] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When calling window.SetContent for a second time, wasm builds will crash.  This seems to happen as a result of trying to delete a texture from the cache.

```
panic: syscall/js: call of Value.Call on null
wasm_exec.js:22 
wasm_exec.js:22 goroutine 809 [running]:
wasm_exec.js:22 syscall/js.Value.Call({{}, 0x7ff8000000000002, 0x0}, {0x1a1202, 0xd}, {0x26a3a50, 0x1, 0x1})
wasm_exec.js:22 	/usr/lib/go/src/syscall/js/js.go:438 +0x3f
wasm_exec.js:22 github.com/fyne-io/gl-js.DeleteTexture(...)
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/github.com/fyne-io/gl-js@v0.1.0/gl_webgl_wasm.go:303
wasm_exec.js:22 fyne.io/fyne/v2/internal/painter/gl.(*xjsContext).DeleteTexture(0x12ad0e0, {{{}, 0x7ff80001000000a1, 0x1a63410}})
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/fyne.io/fyne/v2@v2.6.0-beta1/internal/painter/gl/gl_wasm.go:135 +0x4
wasm_exec.js:22 fyne.io/fyne/v2/internal/painter/gl.(*painter).getTexture.func1()
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/fyne.io/fyne/v2@v2.6.0-beta1/internal/painter/gl/texture.go:53 +0x6
wasm_exec.js:22 fyne.io/fyne/v2/internal/cache.DeleteTextTexturesFor.func1({{{0x170e780, 0x1f}, 0x41600000, {0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, {0x0, ...}}, ...}, ...)
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/fyne.io/fyne/v2@v2.6.0-beta1/internal/cache/texture_common.go:82 +0x11
wasm_exec.js:22 fyne.io/fyne/v2/internal/cache.DeleteTextTexturesFor.(*Map[...]).Range.func2({0xd5820, 0x15eccc0})
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/fyne.io/fyne/v2@v2.6.0-beta1/internal/async/map.go:60 +0x8
wasm_exec.js:22 internal/sync.(*HashTrieMap[...]).iter(0x297cc0, 0x1be5ea0, 0x26a3d38)
wasm_exec.js:22 	/usr/lib/go/src/internal/sync/hashtriemap.go:512 +0x1f
wasm_exec.js:22 internal/sync.(*HashTrieMap[...]).iter(0x297cc0, 0x1be5e00, 0x26a3d38)
wasm_exec.js:22 	/usr/lib/go/src/internal/sync/hashtriemap.go:505 +0x10
wasm_exec.js:22 internal/sync.(*HashTrieMap[...]).iter(0x297cc0, 0x1be5cc0, 0x26a3d38)
wasm_exec.js:22 	/usr/lib/go/src/internal/sync/hashtriemap.go:505 +0x10
wasm_exec.js:22 internal/sync.(*HashTrieMap[...]).Range(0x297cc0, 0x26a3d38)
wasm_exec.js:22 	/usr/lib/go/src/internal/sync/hashtriemap.go:495 +0xe
wasm_exec.js:22 sync.(*Map).Range(0x129d360, 0x26a3d38)
wasm_exec.js:22 	/usr/lib/go/src/sync/hashtriemap.go:115 +0x4
wasm_exec.js:22 fyne.io/fyne/v2/internal/async.(*Map[...]).Range(...)
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/fyne.io/fyne/v2@v2.6.0-beta1/internal/async/map.go:59
wasm_exec.js:22 fyne.io/fyne/v2/internal/cache.DeleteTextTexturesFor({0x2918b0, 0x15e1c20})
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/fyne.io/fyne/v2@v2.6.0-beta1/internal/cache/texture_common.go:79 +0x4
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*window).RescaleContext(0x1400fc0)
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/fyne.io/fyne/v2@v2.6.0-beta1/internal/driver/glfw/window_wasm.go:481 +0x9
wasm_exec.js:22 fyne.io/fyne/v2/internal/async.EnsureMain(0x2695980)
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/fyne.io/fyne/v2@v2.6.0-beta1/internal/async/goroutine.go:46 +0x8
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*window).SetContent(0x1400fc0, {0x28f1f8, 0x156a000})
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/fyne.io/fyne/v2@v2.6.0-beta1/internal/driver/glfw/window.go:246 +0x18
wasm_exec.js:22 main.main.func1()
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne-dev/main.go:48 +0x14
wasm_exec.js:22 fyne.io/fyne/v2/widget.(*Button).Tapped(0x1472e10, 0x26987e0)
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/fyne.io/fyne/v2@v2.6.0-beta1/widget/button.go:188 +0x9
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*window).mouseClickedHandleTapDoubleTap(0x1400fc0, {0x28f258, 0x1472e10}, 0x26987e0)
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/fyne.io/fyne/v2@v2.6.0-beta1/internal/driver/glfw/window.go:584 +0x21
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*window).processMouseClicked(0x1400fc0, 0x1, 0x0, 0x0)
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/fyne.io/fyne/v2@v2.6.0-beta1/internal/driver/glfw/window.go:545 +0xa2
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*window).mouseClicked(0x1400fc0, 0x17fe700, 0x0, 0x0, 0x0)
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/fyne.io/fyne/v2@v2.6.0-beta1/internal/driver/glfw/window_wasm.go:199 +0x1a
wasm_exec.js:22 created by github.com/fyne-io/glfw-js.CreateWindow.func7 in goroutine 10
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/github.com/fyne-io/glfw-js@v0.2.0/browser_wasm.go:225 +0x19
```

### How to reproduce

1. `fyne serve` the example code
2. Open the developer console to see logs
3. Click the button and observe the crash

Repeat for 2.5.4 (no crash, window updates) and 2.6.0-beta1 or develop (crash, window doesn't update)

### Screenshots

_No response_

### Example code

```go
package main

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("crash")

	w.SetContent(
		container.NewVBox(
			widget.NewLabel("Click this button to crash wasm"),
			widget.NewButton("Crash", func() {
				w.SetContent(widget.NewLabel("Done"))
			})))

	w.ShowAndRun()
}
```

### Fyne version

2.6.0-beta1

### Go compiler version

go1.24.1 linux/amd64

### Operating system and version

Arch Linux / Chrome

### Additional Information

_No response_

Trying out the fix in that commit and it no longer reliably crashes, so that does appear to have worked.  I did just get this though during one of my refreshes

```
panic: syscall/js: call of Value.Call on null
wasm_exec.js:22 
wasm_exec.js:22 goroutine 1 [running]:
wasm_exec.js:22 syscall/js.Value.Call({{}, 0x7ff8000000000002, 0x0}, {0x19fdd6, 0x8}, {0x1433b68, 0x4, 0x4})
wasm_exec.js:22 	/usr/lib/go/src/syscall/js/js.go:438 +0x3f
wasm_exec.js:22 github.com/fyne-io/gl-js.Viewport(...)
wasm_exec.js:22 	/home/hayden/Development/Go/pkg/mod/github.com/fyne-io/gl-js@v0.1.0/gl_webgl_wasm.go:824
wasm_exec.js:22 fyne.io/fyne/v2/internal/painter/gl.(*xjsContext).Viewport(0x12cb320, 0x0, 0x0, 0xf00, 0x7fb)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/painter/gl/gl_wasm.go:238 +0x4
wasm_exec.js:22 fyne.io/fyne/v2/internal/painter/gl.(*painter).SetOutputSize(0x15aa000, 0xf00, 0x7fb)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/painter/gl/painter.go:103 +0x6
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.updateGLContext(0x1400fc0)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/loop.go:228 +0x1d
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow.func1()
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/loop.go:197 +0xa
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*window).RunWithContext(0x1400fc0, 0x17b5cf0)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/window.go:899 +0x18
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).repaintWindow(0x15e3d90, 0x1400fc0)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/loop.go:191 +0x4
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).drawSingleFrame(0x15e3d90)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/loop.go:89 +0x1b
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).runGL(0x15e3d90)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/loop.go:167 +0x76
wasm_exec.js:22 fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run(0x15e3d90)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/internal/driver/glfw/driver.go:162 +0x6
wasm_exec.js:22 fyne.io/fyne/v2/app.(*fyneApp).Run(0x15e3e40)
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/fyne/app/app.go:72 +0xf
wasm_exec.js:22 main.main()
wasm_exec.js:22 	/home/hayden/Development/Go/src/github.com/hkparker/wedding/frontend/main.go:403 +0x28
```

Looks to be a similar type of bug somewhere

--- a/internal/driver/glfw/window.go
+++ b/internal/driver/glfw/window.go
@@ -243,7 +243,9 @@ func (w *window) SetContent(content fyne.CanvasObject) {
 	if content != nil {
 		content.Show()
 	}
-	async.EnsureMain(w.RescaleContext)
+	async.EnsureMain(func() {
+		w.RunWithContext(w.RescaleContext)
+	})
 }
 
 func (w *window) Canvas() fyne.Canvas {

